resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1513,7 +1518,11 @@ private[spark] object Client extends Logging {
     }
 
     sys.env.get(ENV_DIST_CLASSPATH).foreach { cp =>
-      addClasspathEntry(getClusterPath(sparkConf, cp), env)
+      val newCp = if (Utils.isTesting) {
```

Comment:
Maybe let's add a comment with mentioning the JIRA number.

Reply:
ok, will add comments after the hadoop2 check  pass